Automattic/wp-calypso
# Pattern Assembler update a11y

## Proposed Changes

1. On PA main screen 
I set tabIndex of p[attern selector -1](https://github.com/Automattic/wp-calypso/pull/67716/files#diff-92e3340cf94e1d29b631265b8f7f4add59a27de23cfbd04818dd01a7c239a956R31) so it won't mess the tab sequence when hidden

2. On PA pattern selector screen
After the pattern selector is shown, it should [focused](https://github.com/Automattic/wp-calypso/pull/67716/files#diff-92e3340cf94e1d29b631265b8f7f4add59a27de23cfbd04818dd01a7c239a956R21) so that user can tab through the top-most items on the list 

3. On PA main screen with selected patterns
- Update visibility of actions bar using opacity so it can be focused
- Make [new component](https://github.com/Automattic/wp-calypso/pull/67716/files#diff-a6539ce815baabe2ceb91b9ef42a5b568c7a2bbdfc82662e41a39245afdc456fR46) for pattern list item to handle their own focus state 

## Testing Instructions

- Using Tab key to navigate through items on PA and use Space/Enter key to choose them

1. On PA main screen 
- Tabbing through the items on screen, it should highlight the items in correct order 
   Back -> Choose a header -> Add section -> Choose a footer
   Recording

   https://user-images.githubusercontent.com/10071857/190944578-5ecfd3ec-d3e9-4385-a150-d209e1ef8a30.mp4

2. On PA pattern selector screen
- Once the pattern selector shows up, press tab key will highlight the top-most pattern in the list
   Recording

   https://user-images.githubusercontent.com/10071857/190944793-caf97718-aa4f-4661-abd8-76d23008902f.mp4

3. On PA main screen with selected patterns
- Upon selecting the patterns on pattern selector, users will be able to iterate through the patterns list & action bar items using Tab key
  Recording

  https://user-images.githubusercontent.com/10071857/190945121-fdb803ce-b760-4d07-b5cf-b5a1788c6a5e.mp4



## Reference

Related to #67632
